scripts/ci/docker-compose/local.yml

```diff
@@ -27,12 +27,6 @@ services:
     # tests (such as logs)
     volumes:
       # START automatically generated volumes from VOLUMES_FOR_SELECTED_MOUNTS in docker_command_utils.py
-      - type: bind
```

Comment:
Those can't be removed. You need to make them work for `breeze` experience. Docker-compose here is NOT EXCLUSIVELY for codespaces. It is there to Power "Breeze" environment, which is the main environment not only for some contributors but also for our CI, So removing those is not good (especially that those entries (as you can see aboeve) are automatically generated based on definition of which volumes shold be mounted (see the comment just above). 

We need to find another way if this faile on devcontainer (but it is strange because it worked before).

Reply:
> Note: The changes in local.yml are just temp and only to get the codespace instance to run, I'm hoping to discuss it further in order to find a better solution

Ah, I only saw that now. Yep. We need to find another way.

Reply:
Actually it's quite likely we do not need to use "local" at all for codespaces. This is only needed for local breeze setup when you want to share multiple folders with your container. And the main reason it is not the whole "airflow" folder, is that by mouting whole "source" directory you share all kinds of artifacts - for example .egginfo files whe  you locally run `pip install -e .` - that often produce various errors when you mount them from host. 

In case of codespaces we ONLY have in-container environment, so we do not need to worry about and instead of using local.yml we could create a separate file that would mound whole "airflow" folder (including the .git you mentioned) .


Actually even better ... we already have such a file :)

* `local-all-sources.yml` is what we should use instead 

This file is used by breeze when you specify `breeze --mount-source all` which is useful in a numebr of situations (mostly when you need the `.git` folder you mentioned.

So I think better solution will be simply to use `local-all-sources.yml` instead.

Reply:
This is great, I will try with the `local-all-sources.yml` but it also contains the same breaking mounts as `local.yml`

      - ../../../.bash_aliases:/root/.bash_aliases:cached
      - ../../../.bash_history:/root/.bash_history:cached
      - ../../../.inputrc:/root/.inputrc:cached
      - ../../../tmp:/tmp:cached

Reply:
And we would still have source code in 2 different locations `/opt/airflow` & `/workspaces/airflow`. I don't know what to make of that!

Reply:
> This is great, I will try with the `local-all-sources.yml` but it also contains the same breaking mounts as `local.yml`

Those mounts are not breaking. Bind mounts are very different from those "old style" mounts. The "old style" mounts will automatically create missing "sources" as empty dirs. Not PERFECT for local development, but good enough for codespaces.

Reply:
> And we would still have source code in 2 different locations `/opt/airflow` & `/workspaces/airflow`. I don't know what to make of that!

No problem whatsoever. There is no performance penalty if the same folders are mounted in two places. We actually often do it in breeze, it makes things easier.

I do not want to make "/workspaces/folder" influence Airflow's dev env. It's the choice made by DevContainers, but from our point of view this is arbitrary choice of just one dev env approaches and we should not follow it. If we could disable it, it would be great (maybe?) but if not - I am not too worried.

Reply:
Actually I looked it up quickly and we can remove `local-sources-all.yml" altogether and use this
https://code.visualstudio.com/remote/advancedcontainers/change-default-source-mount to mount airflow sources to "/opt/airflow". 

Issue solved.

Reply:
Both issues solved :)

Reply:
Did you try it? I haven't tried your suggestions yet.

Because these are only specific to the scenario where you use a dockerfile in your devcontainer.json and not dockercompose file, there are the available options for docker compose https://code.visualstudio.com/docs/remote/devcontainerjson-reference#_docker-compose-specific-properties

Reply:
Now for the specific `workspaceFolder` property, I already tried it by setting it to `/opt/airflow` but it didn't work and still put me in the default `/workspaces/airflow` folder :)

Reply:
Bummer

Reply:
Maybe switch devcontainer files from using docker-compose to dockerfile/image!

Reply:
That will make it difficult for sharing base/postgres/mysql configs between Breeze when they chamge. Those files evolve and it is best if we keep them in sync. Besides for Postgres/MySQL you need to run separate containers for databases and that is way easier with docker-compose.